runtime/src/accounts_db.rs

```diff
@@ -2446,27 +2461,42 @@ impl AccountsDb {
             }
         });
         let dirty_stores_len = dirty_stores.len();
+        let num_threads = self.thread_pool_clean.current_num_threads();
+        let chunk_size = 1.max((dirty_stores_len + num_threads - 1) / num_threads);
         let pubkeys = DashSet::new();
-        timings.oldest_dirty_slot = max_slot.saturating_add(1);
-        for (slot, store) in dirty_stores {
-            timings.oldest_dirty_slot = std::cmp::min(timings.oldest_dirty_slot, slot);
-            store.accounts.account_iter().for_each(|account| {
-                pubkeys.insert(account.meta.pubkey);
-            });
-        }
+        self.thread_pool_clean.install(|| {
```

Comment:
the threadpool here depends on whether we're at startup or not
startup can use all threads

Reply:
Gotcha, I'll change to something like:
```
if startup {
    use all the threads
} else {
    // runtime
    use thread_pool_clean to limit threads
}
```